client/my-sites/comments/controller.js

```diff
@@ -9,10 +9,17 @@ import { includes } from 'lodash';
 /**
  * Internal dependencies
  */
-import route from 'lib/route';
 import CommentsManagement from './main';
+import config from 'config';
+import route from 'lib/route';
 
-const VALID_STATUSES = [ 'pending', 'approved', 'spam', 'trash', 'all' ];
+const VALID_STATUSES = [
+	'pending',
+	'approved',
+	'spam',
+	'trash',
+	...config.isEnabled( 'comments/management/all-list' ) && 'all',
```

Comment:
The spread operator mirrors an `Object.assign`, and that's why I used it in the `CommentNavigation` bulk actions in this same PR.

I tried dropping it for the valid statuses, and I realized that when `isEnabled === true` it works ok.
But, when it's `=== false`, instead of simply not adding `'all'` to the array, it adds a `false` element.
Using the spread operator it works as intended.